# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/WindowGlobalParent.h
# Commit: 09fcee21b246
# Full Hash: 09fcee21b246dbd3622203d214234ee5116805f4
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2019-07-10 21:50:49
# Regressor Bug: 1561395
# File Overlap Count: 2
# Description:
#   Bug 1561395 - Move drawSnapshot API to WindowGlobalParent and allow specifying the whole viewport as a rect. r=mattwoodrow,nika
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D35842
# ==============================================================================

diff -r 0873b1c9bca7 -r 09fcee21b246 dom/ipc/WindowGlobalParent.h
--- a/dom/ipc/WindowGlobalParent.h	Thu Jul 04 00:40:02 2019 +0000
+++ b/dom/ipc/WindowGlobalParent.h	Wed Jul 10 16:45:46 2019 +0000
@@ -8,6 +8,7 @@
 #define mozilla_dom_WindowGlobalParent_h
 
 #include "mozilla/RefPtr.h"
+#include "mozilla/dom/DOMRect.h"
 #include "mozilla/dom/PWindowGlobalParent.h"
 #include "mozilla/dom/BrowserParent.h"
 #include "nsRefPtrHashtable.h"
@@ -21,6 +22,11 @@
 class nsFrameLoader;
 
 namespace mozilla {
+
+namespace gfx {
+class CrossProcessPaint;
+}  // namespace gfx
+
 namespace dom {
 
 class WindowGlobalChild;
@@ -32,6 +38,7 @@
  */
 class WindowGlobalParent final : public WindowGlobalActor,
                                  public PWindowGlobalParent {
+  friend class gfx::CrossProcessPaint;
   friend class PWindowGlobalParent;
 
  public:
@@ -107,6 +114,10 @@
                                                   uint64_t aPendingSwitchId,
                                                   ErrorResult& aRv);
 
+  already_AddRefed<mozilla::dom::Promise> DrawSnapshot(
+      const DOMRect* aRect, double aScale, const nsAString& aBackgroundColor,
+      mozilla::ErrorResult& aRv);
+
   already_AddRefed<Promise> GetSecurityInfo(ErrorResult& aRv);
 
   // Create a WindowGlobalParent from over IPC. This method should not be called
@@ -117,7 +128,7 @@
   // be called after setting the Manager actor.
   void Init(const WindowGlobalInit& aInit);
 
-  nsISupports* GetParentObject();
+  nsIGlobalObject* GetParentObject();
   JSObject* WrapObject(JSContext* aCx,
                        JS::Handle<JSObject*> aGivenProto) override;
 
@@ -140,6 +151,10 @@
 
   void ActorDestroy(ActorDestroyReason aWhy) override;
 
+  void DrawSnapshotInternal(gfx::CrossProcessPaint* aPaint,
+                            const Maybe<IntRect>& aRect, float aScale,
+                            nscolor aBackgroundColor);
+
  private:
   ~WindowGlobalParent();
 